src/test/shell/integration/incompatible_autoload_externally_test.sh

```diff
@@ -106,10 +111,52 @@ local_repository(
 EOF
 }
 
+function mock_apple_support() {
+  apple_support_workspace="${TEST_TMPDIR}/apple_support_workspace"
+  mkdir -p "${apple_support_workspace}/xcode"
+  touch "${apple_support_workspace}/xcode/BUILD"
+  touch "${apple_support_workspace}/WORKSPACE"
+  cat > "${apple_support_workspace}/MODULE.bazel" << EOF
+module(name = "apple_support", repo_name = "build_bazel_apple_support")
+EOF
+  cat > "${apple_support_workspace}/xcode/xcode_version.bzl" << EOF
+def _impl(ctx):
+  pass
+
+xcode_version = rule(
+  implementation = _impl,
+  attrs = {
+    "version": attr.string(),
+  }
+)
+EOF
+
+  cat >> MODULE.bazel << EOF
+bazel_dep(
+    name = "apple_support",
+    repo_name = "build_bazel_apple_support",
+)
+local_path_override(
+    module_name = "apple_support",
+    path = "${apple_support_workspace}",
+)
+EOF
+
+  cat > WORKSPACE << EOF
+workspace(name = "test")
+local_repository(
+    name = "build_bazel_apple_support",
+    path = "${apple_support_workspace}",
+)
+EOF
+}
+
 function test_missing_necessary_repo_fails() {
   # Intentionally not adding apple_support to MODULE.bazel (and it's not in MODULE.tools)
   cat > WORKSPACE << EOF
 workspace(name = "test")
+# TODO: protobuf's WORKSPACE macro has an unnecessary dependency on build_bazel_apple_support.
```

Comment:
This should be fixed at some point, but doesn't matter too much right now as `apple_support` doesn't have any of the Starlarkified rules yet.